modules/common/configmap/configmap.go

```diff
@@ -263,3 +263,98 @@ func GetConfigMap(
 
 	return configMap, ctrl.Result{}, nil
 }
+
+// GetConfigMapListWithLabel - Get all configmaps in namespace of the obj matching label selector
+func GetConfigMapListWithLabel(
+	ctx context.Context,
+	h *helper.Helper,
+	namespace string,
+	labelSelectorMap map[string]string,
+) (*corev1.ConfigMapList, error) {
+	configMaps := &corev1.ConfigMapList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(namespace),
+		client.MatchingLabels(labelSelectorMap),
+	}
+
+	if err := h.GetClient().List(ctx, configMaps, listOpts...); err != nil {
+		err = fmt.Errorf("Error listing config map for labels: %v - %v", labelSelectorMap, err)
+		return nil, err
+	}
+
+	return configMaps, nil
+}
+
+// EnsureOperatorConfigMap - Create ConfigMap consumed by operators
+func EnsureOperatorConfigMap(
+	ctx context.Context,
+	h *helper.Helper,
+	obj client.Object,
+	name string,
+	data map[string]string,
+	labels map[string]string,
+	scope string,
+) error {
+	cmLabels := util.MergeStringMaps(
+		labels,
+		map[string]string{
+			"operators.openstack.org/consumer": "operator",
+			"operators.openstack.org/scope":    scope,
+		},
+	)
+
+	cms := []util.Template{
+		{
+			Name:       name,
+			Namespace:  obj.GetNamespace(),
+			Type:       util.TemplateTypeConfig,
+			CustomData: data,
```

Comment:
> how this data is customized? is the operator supposed to provide this info according to the labels and the scope? It's not clear to me:
> 
>     1. how we determine which is the `scope` and the meaning of this term in this context;
scope represents an operator which consumes this configmap. What I was planning was supporting 'global' (which is presented to all operators) and a specific operator name such as keystone, neutron and etc.

>     2. what `consumer: operator` means: do we have other consumers? The design behind this is not clear;
This was added to represent that this config map is used by operators, instead of actual services.
Each operators should look up key-values stored in these config maps and generate config files for services.
 
>     3. how `data`, which is passed to this function is built: is this data part of the `ini` file that goes within the service config (what we inject via `customServiceConfig` in the CR?)
>        If yes, I think we're violating the agreement that we don't have to change or provide automation to change the service config, because each operator should be seen as a black box where the human operator is supposed to inject pieces of config (via overrides) and provide files via `ExtraMounts`.
This configmap should not be directly used by services but only by operators. Each operator is responsible to manage config files for its own services and an operator should not directly interact with the services managed by the other operators.

Reply:
> > how this data is customized? is the operator supposed to provide this info according to the labels and the scope? It's not clear to me:
> > ```
> > 1. how we determine which is the `scope` and the meaning of this term in this context;
> > ```
> 
> scope represents an operator which consumes this configmap. What I was planning was supporting 'global' (which is presented to all operators) and a specific operator name such as keystone, neutron and etc.

Yeah, we have the same thing here: https://github.com/fmount/lib-common/blob/extra_volumes/modules/storage/storage.go#L32
> 
> > ```
> > 2. what `consumer: operator` means: do we have other consumers? The design behind this is not clear;
> > ```
> 
> This was added to represent that this config map is used by operators, instead of actual services. Each operators should look up key-values stored in these config maps and generate config files for services.

Not 100% sure this is the right approach, used by operators means we need to put, operator by operator, a lot of logic just to make sure the options are consistent and services don't crash because that logic didn't take into account that specific corner case: having the operator choosing the behavior of the component they're going to deploy seems the best option to avoid the issues we did in the past with TripleO. We can give some guidance providing "samples", but AFAIK that's the direction, especially in storage where you have to deal w/ multiple backends.

> 
> > ```
> > 3. how `data`, which is passed to this function is built: is this data part of the `ini` file that goes within the service config (what we inject via `customServiceConfig` in the CR?)
> >    If yes, I think we're violating the agreement that we don't have to change or provide automation to change the service config, because each operator should be seen as a black box where the human operator is supposed to inject pieces of config (via overrides) and provide files via `ExtraMounts`.
> > ```
> 
> This configmap should not be directly used by services but only by operators. Each operator is responsible to manage config files for its own services and an operator should not directly interact with the services managed by the other operators.

See my previous comment that should cover this part as well